src/globus_sdk/scopes.py

```diff
@@ -22,6 +22,22 @@ def __init__(
             for scope_name in known_scopes:
                 setattr(self, scope_name, self.urn_scope_string(scope_name))
 
+    # custom __getattr__ instructs `mypy` that unknown attributes of a ScopeBuilder are
+    # of type `str`, allowing for dynamic attribute names
+    # to test, try creating a module with
+    #
+    #       from globus_sdk.scopes import TransferScopes
+    #       x = TransferScopes.all
+    #
+    # the assignment to `x` will fail type checking without this because `all` will be
+    # unknown to `mypy`
```

Comment:
A small nitpick/wording suggestion that I think is more clear:
```suggestion
    # without this method, the assignment to `x` would fail type checking
    # because `all` is unknown to mypy
```

Reply:
Thanks; I agree that the reword is better. Committed!